tests/functional/database-management.test.js

```diff
@@ -1,119 +1,269 @@
+                'extraKeysTotal': {value: calculateTotalSize(calculateKeySize(testParams.extraKeysKeySize), testParams.extraKeysValueSize) * testParams.numberOfExtraKeys},
+                'largeKeyTotal': {value: calculateTotalSize(calculateKeySize(testParams.largeKey), testParams.largeKeyValueSize)}
+            });
+
+            before('createDB of size', async function () {
+                await this.api.createDB(testParams.databaseSize)
+            });
+
+            it(`should show remainingBytes of ${testParams.databaseSize}`, async function () {
+                expect(await this.api.size()).to.deep.include({remainingBytes: testParams.databaseSize});
+            });
+
+            it('should show bytes of 0', async function () {
+                expect(await this.api.size()).to.deep.include({bytes: 0});
+            });
+
+            it('should show keys of 0', async function () {
+                expect(await this.api.size()).to.deep.include({keys: 0});
+            });
+
+            context('creating a key', function () {
 
-            context('with empty db', function () {
 
-                it('should be able to hasDB', async function () {
-                    expect(await this.api.hasDB()).to.be.true;
+                before(`create key of value size ${testParams.initialKeyValueSize}`, async function () {
+                    await this.api.create(testParams.initialKey, generateString(testParams.initialKeyValueSize));
                 });
 
-                it('should be able to deleteDB', async function () {
-                    await this.api.deleteDB();
+                it(`should show remainingBytes of ${testParams.databaseSize - testParams.initialKeyTotal}`, async function () {
+                    expect(await this.api.size()).to.deep.include({remainingBytes: testParams.databaseSize - testParams.initialKeyTotal});
                 });
 
-                keysAndSizeShouldReturnZero();
+                it(`should show bytes of ${testParams.initialKeyTotal}`, async function () {
+                    expect(await this.api.size()).to.deep.include({bytes: testParams.initialKeyTotal});
+                });
 
+                it('should show keys of 1', async function () {
+                    expect(await this.api.size()).to.deep.include({keys: 1});
+                });
+            });
 
-                context('should throw errors', function () {
+            context('creating more keys', function () {
 
-                    it('when attempting to createDB', async function () {
-                        await this.api.createDB().should.be.rejectedWith('DATABASE_EXISTS');
-                    });
+                before(`creating ${testParams.numberOfExtraKeys} more keys`, async function () {
+                    // createKeys creates keys as create(base_string + i, 'value'), so a total size of base_string + 1 + 5. Passing in a single byte string will result in a 2 byte key.
```

Comment:
1. This statement only holds true for `i` (number of keys) less than 10, thus this test isn't as flexible as implied.
2. This comment should be more tightly coupled with the `testParams` IMO, as a decent chunk of it is magic numbers without this correlation.

An alteration could be made to `createKeys` so that no magic numbers or explanations are necessary. Simple get it to return to the caller the list of keys (perhaps object of keys -> values) created by the helper. It is then easy to calculate the bytes expected to be used with no magic.

Reply:
Good idea! Refactored as suggested